library/src/data/index.js

```diff
@@ -1 +1,24 @@
-window.CoNDeT.data = {}
\ No newline at end of file
+window.CoNDeT.data = {
+  State: (function () {
+    function constructor() {}
+
+    constructor.prototype.state = [];
+    constructor.prototype.subscribers = [];
+
+    constructor.prototype.methods = {
+      subscribe: function (fun) {
+        this.subscribers.push(fun);
+      },
+      set: function (state) {
+        this.state = state;
+        this.call_subscribers();
+      },
+      call_subscribers: function () {
+        this.subscribers.forEach((element) => {
+          element.call();
+        });
+      },
```

Comment:
Nie potrzeba tutaj używać `call`. `call` jest dość niewydajne, bo podmienia `this` na pierwszy argument. Dodatkowo nie podajesz obecnego state. Nie używamy też arrow functions ze względu na wymagania projektu, bo nie jest wspierane w IE
```
var self = this;
this.subscribers.forEach(function (cb) {
  cb(self.state);
});
```